types/block.go

```diff
@@ -1180,6 +1180,13 @@ func (msgs *Messages) SortMessages() {
 	})
 }
 
+// IsSorted returns whether the messages are sorted by namespace id
+func (msgs *Messages) IsSorted() bool {
+	return sort.SliceIsSorted(msgs.MessagesList, func(i, j int) bool {
+		return bytes.Compare(msgs.MessagesList[i].NamespaceID, msgs.MessagesList[j].NamespaceID) < 0
```

Comment:
Should this be

```suggestion
		return bytes.Compare(msgs.MessagesList[i].NamespaceID, msgs.MessagesList[j].NamespaceID) <= 0
```

?

Reply:
That's a very good question. I think yes but will confirm and add unit tests to cover this scenario.

Reply:
thanks for raising this question. We want `Less` to return `false` here because

```
	// If both Less(i, j) and Less(j, i) are false,
	// then the elements at index i and j are considered equal.
```

and since `bytes.Compare([]byte{1, 2, 3, 4, 5, 6, 7, 8}, []byte{1, 2, 3, 4, 5, 6, 7, 8}) == 0`, we want the conditional to be `0 < 0` which will evaluate to false.


So although the comparison doesn't need `<=`, the previous implementation was confusing b/c it implemented two distinct comparison functions ([here](https://github.com/celestiaorg/celestia-core/blob/70bd95acc1bdbf0c7240c63853cca78b332577c2/types/block.go#L1179) and [here](https://github.com/celestiaorg/celestia-core/blob/70bd95acc1bdbf0c7240c63853cca78b332577c2/types/block.go#L1186)). These comparison functions should be identical so I refactored to extract `ByNamespace`.

Ref: 
1. https://pkg.go.dev/sort#Interface
1. https://pkg.go.dev/sort#IsSorted
1. https://yourbasic.org/golang/how-to-sort-in-go/#sort-custom-data-structures

Reply:
Neat! I see you added a test for equal namespaces, but can you also document what you wrote here?

Reply:
Done in https://github.com/celestiaorg/celestia-core/pull/825/commits/a7210889e6a0f05645bda5605d49a7614c92430b